modules/serverless-tag-to-digest-resolution.adoc

```diff
@@ -0,0 +1,52 @@
+// Module included in the following assemblies
+//
+// * serverless/admin_guide/serverless-configuration.adoc
+
+:_content-type: CONCEPT
+[id="serverless-tag-to-digest-resolution_{context}"]
+= Configuring tag-to-digest resolution
+
+If the Knative Serving controller has access to the container registry, Knative Serving resolves image tags to a digest when you create a revision of a service. This is referred to as _tag-to-digest resolution_, and helps to provide consistency for deployments.
+
+If your container registry uses a self-signed certificate, you must configure the Knative Serving controller so that it trusts that certificate. To do this, you must create a config map or secret and then configure controller custom certificates by modifying the `controller-custom-certs` spec in the `KnativeServing` custom resource (CR). The config map or secret must reside in the same namespace as the Knative Serving CR. If no controller custom certificate is specified, this setting defaults to use the `config-service-ca` config map.
+
+[id="knative-serving-controller-custom-certs-secrets_{context}"]
+== Using a secret
+
+If the type configured in the `controller-custom-certs` spec is `secret`, the {ServerlessOperatorName} takes no action. The secret is mounted as a secret volume, and Knative components consume the secret directly, assuming that the secret contains the required certificates.
+// example YAML?
+
+[id="knative-serving-controller-custom-certs-configmap_{context}"]
+== Using a config map
+
+If the type configured in the `controller-custom-certs` spec is `ConfigMap`, a config map name must be specified. When the name is specified, this triggers the {ServerlessOperatorName} to:
+
+. Create and mount a volume containing the certificate in the controller.
+. Set the required environment variable properly.
+
+If no config map name is specified, the {ServerlessOperatorName} takes no action.
+
+The following example uses a certificate in a config map named `certs` in the `knative-serving` namespace:
+
+.Example KnativeServing CR
+[source,yaml]
+----
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  controller-custom-certs:
+    name: config-service-ca
+    type: ConfigMap <1>
+----
+
```

Comment:
Interestingly, if the user doesn't specify anything from the above yaml it will work as well. The important part is really the PKI configuration below and the Serverles Operator and Knative Operator will do the magic automatically. I'm wondering if we want to leave it here anyway or mention this fact in some way. IMO, the important part of this documentation is really using the "secret".

Reply:
After discussion with @maschmid , I think we need to make it clear that using `type: ConfigMap` and `name: XXX` effectively does nothing. It will only rename the config map but this config map is changed automatically by the operator (it injects the certs into it). So, users can't define their own ConfigMap with secrets - it will be overwritten. For users, changing this KnativeServing configuration only makes sense if they want to use `Secret` type.